src/sagas/reown.js

```diff
@@ -602,6 +653,52 @@ const promptHandler = (dispatch) => (request, requestMetadata) =>
           requestMetadata,
         ));
       } break;
+      case TriggerTypes.SendTransactionConfirmationPrompt: {
+        const sendTransactionResponseTemplate = (accepted) => (data) => resolve({
+          type: TriggerResponseTypes.SendTransactionConfirmationResponse,
+          data: {
+            accepted,
+            payload: data,
+          }
+        });
+
+        dispatch(showSendTransactionModal(
+          sendTransactionResponseTemplate(true),
+          sendTransactionResponseTemplate(false),
+          request.data,
+          requestMetadata
+        ));
+      } break;
+      case TriggerTypes.SendTransactionLoadingTrigger:
+        dispatch(setSendTxStatusLoading());
+        // Use the data from the request instead of getting current state
+        dispatch(setReownModal({
+          show: true,
+          type: ReownModalTypes.SEND_TRANSACTION,
+          data: {
+            ...(request.data || {}), // Use data from the request
+            isLoading: true,
+            isSuccess: false,
+            isError: false
+          }
+        }));
```

Comment:
Why do you need to set the model here and in the finished trigger? The others you don't need this.